### ISSUE TITLE
Island borders shrunk, error on latest ci builds

### ISSUE THREAD

**Issue by spookymgmt** [BentoBoxWorld/Border#123]
### Expected behavior

No error no island shrink

### Observed/Actual behavior

a

### Steps/models to reproduce

a

### BentoBox version

on latest aoneblock & bentobox ci. builds

```[16:13:34] [Server thread/ERROR]: Could not pass event PlayerMoveEvent to BentoBox v2.3.0
java.lang.IllegalArgumentException: Direction's magnitude is 0!
    at com.google.common.base.Preconditions.checkArgument(Preconditions.java:145) ~[guava-31.1-jre.jar:?]
    at org.bukkit.util.BoundingBox.rayTrace(BoundingBox.java:838) ~[axolotl-api-1.20.1-R0.1-SNAPSHOT.jar:?]
    at world.bentobox.border.listeners.PlayerListener.lambda$onPlayerLeaveIsland$10(PlayerListener.java:154) ~[?:?]
    at java.util.Optional.ifPresent(Optional.java:178) ~[?:?]
    at world.bentobox.border.listeners.PlayerListener.onPlayerLeaveIsland(PlayerListener.java:151) ~[?:?]
    at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor1951.execute(Unknown Source) ~[?:?]
    at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77) ~[axolotl-api-1.20.1-R0.1-SNAPSHOT.jar:?]
    at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:77) ~[axolotl-api-1.20.1-R0.1-SNAPSHOT.jar:git-Axolotl-"3841885"]
    at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[axolotl-api-1.20.1-R0.1-SNAPSHOT.jar:?]
    at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[axolotl-1.20.1.jar:git-Axolotl-"3841885"]
    at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[axolotl-1.20.1.jar:git-Axolotl-"3841885"]
    at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:617) ~[axolotl-api-1.20.1-R0.1-SNAPSHOT.jar:?]
    at net.minecraft.server.network.ServerGamePacketListenerImpl.handleMovePlayer(ServerGamePacketListenerImpl.java:1670) ~[?:?]
    at net.minecraft.network.protocol.game.ServerboundMovePlayerPacket.handle(ServerboundMovePlayerPacket.java:29) ~[?:?]
    at net.minecraft.network.protocol.game.ServerboundMovePlayerPacket$Pos.handle(ServerboundMovePlayerPacket.java:66) ~[?:?]
    at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:51) ~[?:?]
    at net.minecraft.server.TickTask.run(TickTask.java:18) ~[axolotl-1.20.1.jar:git-Axolotl-"3841885"]
    at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
    at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
    at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1364) ~[axolotl-1.20.1.jar:git-Axolotl-"3841885"]
    at net.minecraft.server.MinecraftServer.wrapRunnable(MinecraftServer.java:199) ~[axolotl-1.20.1.jar:git-Axolotl-"3841885"]
    at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
    at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1341) ~[axolotl-1.20.1.jar:git-Axolotl-"3841885"]
    at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1334) ~[axolotl-1.20.1.jar:git-Axolotl-"3841885"]
    at net.minecraft.util.thread.BlockableEventLoop.runAllTasks(BlockableEventLoop.java:114) ~[?:?]
    at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1450) ~[axolotl-1.20.1.jar:git-Axolotl-"3841885"]
    at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1196) ~[axolotl-1.20.1.jar:git-Axolotl-"3841885"]
    at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:326) ~[axolotl-1.20.1.jar:git-Axolotl-"3841885"]
    at java.lang.Thread.run(Thread.java:833) ~[?:?]```

### Plugin list

_No response_

### Other

_No response_

**Comment by tastybento** [comment]
Can you share what actually happened? I see the error, but what happened before that? You mention the island shrank?

**Comment by tastybento** [comment]
I'd be very interested to see if in the logs this is written anywhere:

`Island protection range changed from (some number) to (some number) for (player name) due to permission.`

where the items in brackets are numbers or the player name. Can you check your logs, please?

**Comment by spookymgmt** [comment]
Here's the full log https://mclo.gs/99hNGTZ

And basically, last night, I added the new ci. aoneblock, and did not update bento box. 

Server rebooted, and the aoneblock addon didnt load. This allowed players to grief others islands.


Fast forward to this morning, I updated bento box to 2.3, and rolled back 

- island world
- spawn world playerdata
- playervaults
- auctionhouse data
- essentials data

(thats it)

then I started it up, and noticed that all borders were a 1x1. When a player joined, i teleported to them, and yeah the border was just 1x1.

Then I reverted aoneblock & bentobox to previous version, and restored bento data, and all was solved.

**Comment by tastybento** [comment]
Please update to 4.2.2 release, according to the log you are on 4.2.1. The error should be fixed in 4.2.2, however, I don't know how the island sizes could have been set to 0.

The log is huge, and it has that error, but I'd like to see the log from the time when aoneblock didn't load to try and work out why it didn't load.

**Comment by spookymgmt** [comment]
This is solved on latest border plugin. Thank you so much
